### DISCUSSION TITLE
Making a version for S805/S802 boards

### DISCUSSION THREAD

**Discussion by SadarSSI** [devmfc/debian-on-amlogic#3] [Q&A]
Hi, it's me again, 

I would like to use your work to make a version for meson-gmx SOC 805/802 boards.

In fact the armbian versions I find are with a kernel 3.10 which works quite well or 5.x which has some bugs.

I'd appreciate it if you could share your know-how to create a Debian version for this board, which has a 32-bit SOC.

On the Debian repositories you'll find the DTB for my board: meson8-minix-neo-x8.dtb and other DTBs work with ARMBIAN like meson8m2-mxiii-plus.dtb.

Do you think this is possible? 

I don't have a lot of experience compiling linux, but I can work very well with docker or do an armhf debootstrap with qemu-system-arm. I also sometimes have to compile a few programs by installing build-essentials and the necessary libraries.

For information, there are sources available for this type of card at https://github.com/xdarklight/linux/tree/meson-mx-integration-6.3-20230410.

So, I think a compilation should work

Translated with www.DeepL.com/Translator (free version)

**Comment by devmfc** [comment]
I gave this some thought over the weekend, but I'm sorry, I think I will never return to 32 bit. 

You already found the Github of xdarklight/Martin, he is one of the mainline maintainers for Amlogic/meson and also the older meson8 SOC's

Hexdump also has some images for meson8, I have never tried his images, but it is worth a try I think. See [https://github.com/hexdump0815/imagebuilder/releases/tag/230224-03]

**Comment by SadarSSI** [comment]
Thank you for your reply. 

I did manage to compile a kernel the sources found on the xdarklight/Martin link. This compilation gives me a uImage that I've put in /boot formatted in FAT. But I think it's necessary to complete this with mkimage commands to finalize the boot and I can't really find a tuto guide for this, so I was asking if you could share this so I can follow a method that works today

One last question: do you think it's possible to test the disk image via qemu before writing it to the SD card? 
I'd like to develop a boot that works with qemu and finalize its development on the SOC: I have a serial port access that works so I can see the boot sequence.

In any case, thank you for your answers.
